File: backend/src/main/java/es/maryshopping/backend/customers/shared/domain/customer/Customer.java
package es.maryshopping.backend.customers.shared.domain.customer;

import lombok.Builder;

@Builder
public record Customer(
        CustomerId id,
        FirstName firstName,
        LastName lastName,
        NationalIdentificationNumber dni,
        BillingAddress billingAddress,
        ShippingAddress shippingAddress,
        PhoneNumber phoneNumber,
        EmailAddress emailAddress
) {
}

File: frontend/app/models/customers/Customer.ts
// 👤 CLIENTE (Estructura de datos)
// Define cómo es un cliente:
// - id, nombre, email, teléfono, dni
// - dirección de envío y facturación
// Usado para registros y perfiles

export interface Address {
  street: string;
  city: string;
  province: string;
  postalCode: string;
  country: string;
}

export interface Customer {
  id: string;
  firstName: string;
  lastName: string;
  dni: string;
  phoneNumber: string;
  emailAddress: string;
  billingAddress: Address;
  shippingAddress: Address;
}
